### PR TITLE
update OP tokens and prices lookups in spellbook

### PR DESCRIPTION
**Pull request by MSilb7**
Brief comments on the purpose of your changes:
Accidentally closed my previous PR https://github.com/duneanalytics/spellbook/pull/1726

Add erc20, nft tokens, prices to Optimism in spellbook.

Notes:

- Added a 'standard' column for nft tokens (does this need to be dropped and re-added?)
- Updated the OP token to have the Optimism token address in prices. This is different than the other network tokens, but it's an erc20 and has an actual contract address.
- Left some tokens commented out in prices if they didn't exist in coinpaprika. Did it this way so it's easier to add if/when coinpaprika expands their tokenlist / adds more L2 tokens, but not sure if that will cause issues (feel free to delete those comments if it does).
- Added a "lower" case in the prices table for contracts. This makes it easier /dummy-proofs when copy/pasting addresses since sometimes etherscan doesn't give you lowercase. But if there's reason not to do that, feel free to modify.

*For Dune Engine V2*
I've checked that:
General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)

Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by augustog** [comment]
There are three changes here:
- Added several tokens to coinpaprika price feeds ⚠️  a handful of those return `is_active = False`, list below.
- Added metadata for 14 new tokens on optimism (github diff got a bit confused with the amount of lines changed), looks good ✅ 
- New column added to nft metadata (@jeff-dude do we have any consideration before merging this?) plus a few additional nft projects (spot checked & looks good)

Tokens to add to coinpaprika (https://coinpaprika.com/add/ ) -- I'd suggest taking them out of this PR

coin_id | is_active | symbol
-- | -- | --
reth-rocket-pool-eth | FALSE | RETH
gohm-governance-ohm | FALSE | GOHM
seur-seur | FALSE | SEUR
socks-unisocks | FALSE | SOCKS
ib-iron-bank | FALSE | IB
xchf-cryptofranc | FALSE | XCHF
premia-premia | FALSE | PREMIA

**Comment by jeff-dude** [comment]
> There are three changes here:
> 
> * Added several tokens to coinpaprika price feeds ⚠️  a handful of those return `is_active = False`, list below.
> * Added metadata for 14 new tokens on optimism (github diff got a bit confused with the amount of lines changed), looks good ✅
> * New column added to nft metadata (@jeff-dude do we have any consideration before merging this?) plus a few additional nft projects (spot checked & looks good)
> 
> Tokens to add to coinpaprika (https://coinpaprika.com/add/ ) -- I'd suggest taking them out of this PR
> 
> coin_id	is_active	symbol
> reth-rocket-pool-eth	FALSE	RETH
> gohm-governance-ohm	FALSE	GOHM
> seur-seur	FALSE	SEUR
> socks-unisocks	FALSE	SOCKS
> ib-iron-bank	FALSE	IB
> xchf-cryptofranc	FALSE	XCHF
> premia-premia	FALSE	PREMIA

i *believe* our dbt command for running views will always do ```CREATE OR REPLACE``` but i'd recommend watching dbt cloud as we merge this to ensure as expected. otherwise, we will just add a ```--full-refresh``` as needed on the view(s)

agreed with removing the rows which coinpaprika has marked as inactive -- can always add later if they get reactivated

**Comment by MSilb7** [comment]
> > There are three changes here:
> > 
> > * Added several tokens to coinpaprika price feeds ⚠️  a handful of those return `is_active = False`, list below.
> > * Added metadata for 14 new tokens on optimism (github diff got a bit confused with the amount of lines changed), looks good ✅
> > * New column added to nft metadata (@jeff-dude do we have any consideration before merging this?) plus a few additional nft projects (spot checked & looks good)
> > 
> > Tokens to add to coinpaprika (https://coinpaprika.com/add/ ) -- I'd suggest taking them out of this PR
> > coin_id	is_active	symbol
> > reth-rocket-pool-eth	FALSE	RETH
> > gohm-governance-ohm	FALSE	GOHM
> > seur-seur	FALSE	SEUR
> > socks-unisocks	FALSE	SOCKS
> > ib-iron-bank	FALSE	IB
> > xchf-cryptofranc	FALSE	XCHF
> > premia-premia	FALSE	PREMIA
> 
> i _believe_ our dbt command for running views will always do `CREATE OR REPLACE` but i'd recommend watching dbt cloud as we merge this to ensure as expected. otherwise, we will just add a `--full-refresh` as needed on the view(s)
> 
> agreed with removing the rows which coinpaprika has marked as inactive -- can always add later if they get reactivated

yeah +1 on whatever works best to get active tokens in now! We can come back to the others later.

File: models/prices/prices_tokens.sql [hunk 1]
```diff
@@ -1031,10 +1031,90 @@ VALUES
...
+    ("link-chainlink","optimism","LINK","0x350a791bfc2c21f9ed5d10980dad2e2638ffa7f6",18),
+    ("rgt-rari-governance-token","optimism","RGT","0xb548f63d4405466b36c0c0ac3318a22fdcec711a",18),
+    ("mkr-maker","optimism","MKR","0xab7badef82e9fe11f6f33f87bc9bc2aa27f2fcb5",18),
+    ("uni-uniswap","optimism","UNI","0x6fd9d7ad17242c41f7131d257212c54a0e816691",18),
+    ("rai-rai-reflex-index","optimism","RAI","0x7fb688ccf682d58f86d7e38e03f9d22e7705448b",18),
+    ("susd-susd","optimism","sUSD","0x8c6f28f2f1a3c87f0f938b96d27520d9751ec8d9",18),
+    ("seth-seth","optimism","sETH","0xe405de8f52ba7559f9df3c368500b6e6ae6cee49",18),
+    -- ("NULL","optimism","sLINK","0xc5db22719a06418028a40a9b5e9a7c02959d0d08",18),
+    -- ("NULL","optimism","sBTC","0x298b9b95708152ff6968aafd889c6586e9169f1d",18),
+    ("reth-rocket-pool-eth","optimism","rETH","0x9bcef72be871e61ed4fbbc7630889bee758eb81d",18),
```
Comment (augustog) [inline models/prices/prices_tokens.sql:1051]: remove or comment out tokens with inactive price feeds on coinpaprika

**Review by augustog — COMMENTED** [review]
okay so removing those would be enough, the rest looks good. 
I'll coordinate so we merge safely.

**Comment by MSilb7** [comment]
Also, while this is open - it looks like wstETH prices aren't pulling through on L1

`("wsteth-wrapped-liquid-staked-ether-20", "ethereum", "wstETH", "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0", 18),`

File: models/prices/prices_tokens.sql (hunk 1 above)
Comment (MSilb7) [reply models/prices/prices_tokens.sql:1051]: on it

**Comment by MSilb7** [comment]
removed inactive tokens

**Comment by jeff-dude** [comment]
> Also, while this is open - it looks like wstETH prices aren't pulling through on L1
> 
> `("wsteth-wrapped-liquid-staked-ether-20", "ethereum", "wstETH", "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0", 18),`

hmm returns valid on coinpaprika, we'll have to look closer at it

**Comment by jeff-dude** [comment]
> > There are three changes here:
> > 
> > * Added several tokens to coinpaprika price feeds ⚠️  a handful of those return `is_active = False`, list below.
> > * Added metadata for 14 new tokens on optimism (github diff got a bit confused with the amount of lines changed), looks good ✅
> > * New column added to nft metadata (@jeff-dude do we have any consideration before merging this?) plus a few additional nft projects (spot checked & looks good)
> > 
> > Tokens to add to coinpaprika (https://coinpaprika.com/add/ ) -- I'd suggest taking them out of this PR
> > coin_id	is_active	symbol
> > reth-rocket-pool-eth	FALSE	RETH
> > gohm-governance-ohm	FALSE	GOHM
> > seur-seur	FALSE	SEUR
> > socks-unisocks	FALSE	SOCKS
> > ib-iron-bank	FALSE	IB
> > xchf-cryptofranc	FALSE	XCHF
> > premia-premia	FALSE	PREMIA
> 
> i _believe_ our dbt command for running views will always do `CREATE OR REPLACE` but i'd recommend watching dbt cloud as we merge this to ensure as expected. otherwise, we will just add a `--full-refresh` as needed on the view(s)
> 
> agreed with removing the rows which coinpaprika has marked as inactive -- can always add later if they get reactivated

confirmed just a regular dbt run for config:view does in fact replace the old, so ```standard``` column addition was no problem

**Pull request merged**